tests/scripts/thread-cert/Cert_5_1_03_RouterAddressReallocation.py

```diff
@@ -72,113 +96,147 @@ def test(self):
+                   must_next()
+
+        pkts.filter_wpan_src64(ROUTER_1).\
+            filter_LLARMA().\
+            filter_mle_cmd(MLE_PARENT_REQUEST).\
+            filter(lambda p: {
+                              CHALLENGE_TLV,
+                              MODE_TLV,
+                              SCAN_MASK_TLV,
+                              VERSION_TLV
+                              } <= set(p.mle.tlv.type) and\
+                   p.mle.tlv.scan_mask.r == 1 and\
+                   p.ipv6.hlim == 255).\
+            must_next()
+
+        # Step 7: Router_1 sends a MLE Child ID Request to Router_2.
+        #         The following TLVs MUST be present in the MLE Child ID Request:
+        #             - Link-layer Frame Counter TLV
+        #             - Mode TLV
+        #             - Response TLV
+        #             - Timeout TLV
+        #             - TLV Request TLV
+        #             - Version TLV
+        #             - MLE Frame Counter TLV (optional)
+        #         The following TLV MUST NOT be present in the MLE Child ID Request:
+        #             - Address Registration TLV
+
+        _pkt = pkts.filter_wpan_src64(ROUTER_1).\
+            filter_wpan_dst64(ROUTER_2).\
+            filter_mle_cmd(MLE_CHILD_ID_REQUEST).\
+            filter(lambda p: {
+                              LINK_LAYER_FRAME_COUNTER_TLV,
+                              MODE_TLV,
+                              RESPONSE_TLV,
+                              TIMEOUT_TLV,
+                              TLV_REQUEST_TLV,
+                              VERSION_TLV
+            } <= set(p.mle.tlv.type)).\
+                   must_next()
+        _pkt.must_not_verify(lambda p: (ADDRESS_REGISTRATION_TLV) in p.mle.tlv.type)
+
+        # Step 8: Router_1 sends an Address Solicit Request.
+        #         Ensure the Address Solicit Request is properly formatted:
+        #         CoAP Request URI
+        #             coap://<leader address>:MM/a/as
+        #         CoAP Payload
+        #             - MAC Extended Address TLV
+        #             - Status TLV
+        #             - RLOC16 TLV
+
+        _pkt = pkts.filter_wpan_src64(ROUTER_1).\
+            filter_wpan_dst16(ROUTER_2_RLOC16).\
+            filter_coap_request(ADDR_SOL_URI).\
+            filter(lambda p: {
+                              NL_MAC_EXTENDED_ADDRESS_TLV,
+                              NL_STATUS_TLV,
+                              NL_RLOC16_TLV
+                              } <= set(p.coap.tlv.type) and\
+                   p.thread_address.tlv.rloc16 ==
+                   _pkt_as.thread_address.tlv.rloc16).\
+           must_next()
+
+        # Step 9: Router_2 automatically sends an Address Solicit Response.
+
+        pkts.filter_wpan_src64(ROUTER_2).\
+            filter_wpan_dst16(_pkt.wpan.src16).\
+            filter_coap_ack(ADDR_SOL_URI).\
+            filter(lambda p: p.thread_address.tlv.rloc16 ==
```

Comment:
better to add check on status tlv which should be success.

Reply:
added checks